### PR TITLE
Add version 2.1.0 to 'aspect' package

### PR DESCRIPTION
**Pull request by gassmoeller**
This adds the latest release to the 'aspect' package.

@tjhei: We should add updating the spack package to our release-tasklist. Will do so in a follow-up.

### CODE REVIEW

File: var/spack/repos/builtin/packages/aspect/package.py [hunk 1]
```diff
@@ -11,12 +11,13 @@ class Aspect(CMakePackage):
     Earth's mantle and elsewhere."""
 
     homepage = "https://aspect.geodynamics.org"
-    url      = "https://github.com/geodynamics/aspect/releases/download/v2.0.0/aspect-2.0.0.tar.gz"
+    url      = "https://github.com/geodynamics/aspect/releases/download/v2.1.0/aspect-2.1.0.tar.gz"
     git      = "https://github.com/geodynamics/aspect.git"
 
     maintainers = ['tjhei']
 
     version('develop', branch='master')
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/aspect/package.py]: You can actually rename this to master now if you want.

File: var/spack/repos/builtin/packages/aspect/package.py [hunk 2]
```diff
@@ -11,12 +11,13 @@ class Aspect(CMakePackage):
...
+    version('2.1.0', sha256='bd574d60ed9df1f4b98e68cd526a074d0527c0792763187c9851912327d861a3')
     version('2.0.1', sha256='0bf5600c42afce9d39c1d285b0654ecfdeb0f30e9f3421651c95f54ca01ac165')
     version('2.0.0', sha256='d485c07f54248e824bdfa35f3eec8971b65e8b7114552ffa2c771bc0dede8cc0')
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/aspect/package.py]: `setup_environment` needs to be replaced by `setup_run_environment`

File: var/spack/repos/builtin/packages/aspect/package.py [hunk 3]
```diff
@@ -35,5 +36,5 @@ def cmake_args(self):
             ('ON' if '+fpe' in self.spec else 'OFF')
         ]
 
-    def setup_environment(self, spack_env, run_env):
+    def setup_run_environment(self, spack_env, run_env):
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/aspect/package.py]:
```suggestion
    def setup_run_environment(self, env):
```

File: var/spack/repos/builtin/packages/aspect/package.py [hunk 4]
```diff
@@ -35,5 +36,5 @@ def cmake_args(self):
...
         run_env.set('Aspect_DIR', self.prefix)
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/aspect/package.py]:
```suggestion
        env.set('Aspect_DIR', self.prefix)
```

**Comment by gassmoeller** [comment]
Ah, sorry, I did not check the new API.

Concerning the development version name: I think I would prefer to keep it this way for now, I do not know if that name was chosen by @tjhei for a reason, and our users might already use it. Was there a limitation in previous versions that prevented `master` from being a valid version name?

Thanks for the review.

**Comment by gassmoeller** [comment]
Some of the build tests failed, but I could not quite figure out what is happening, it does not seem related to this package. Do the tests try to build the 'aspect' package, or just spack in general?

**Comment by adamjstewart** [comment]
The build tests try to build a set of 5 packages (a mix of Autotools, CMake, Python, etc). The reason the build tests are failing is that ftpmirror.gnu.org is currently down, so it's unable to download the dependencies. Previously we had a download cache that stored these files, but we were seeing problems with it so we removed them. So now every PR is failing its build tests and can't be merged until either ftpmirror.gnu.org comes back or we change the download URL for those packages. I'll start a PR to do the latter.

**Comment by adamjstewart** [comment]
> Concerning the development version name: I think I would prefer to keep it this way for now, I do not know if that name was chosen by @tjhei for a reason, and our users might already use it. Was there a limitation in previous versions that prevented master from being a valid version name?

Yes, before #1983 was merged, `develop` was the only version that sorted higher than numeric releases, and `master` was considered to be the oldest release. Now that #1983 has been merged, `master` and `develop` are treated similarly.

**Pull request merged**